### PR TITLE
Improve/correct list access. Fix memory leaks in tests. Fix memory le…

### PR DESCRIPTION
**Pull request by manderiasian**
…ak in subscription.c:get_Client_Subscriptions()

### CODE REVIEW

File: src/subscription.c [hunk 1]
```diff
@@ -117,21 +129,37 @@ void filter_clients_and_send(wrp_msg_t *wrp_event_msg)
...
-        if(strcmp(sub->service_name, service_name) == 0)
-        {
-            rebar_ll_remove( g_sub_list, node );
-            match_found = true;
-        }
-        node = node->next;
+    rebar_ll_node_t *node;
+    bool found_match;
+
+    node = rebar_ll_find( g_sub_list, compare_and_delete_private_data,
```
[the remark this answers is not in the document]
Comment (manderiasian) [reply src/subscription.c]: I didn't realize a "service-name" can be associated with more than one node, I will correct the code to find delete all nodes that match that service-name.

File: src/crud_tasks.c [hunk 2]
```diff
@@ -151,8 +151,13 @@ int processCrudRequest( wrp_msg_t *reqMsg, wrp_msg_t **responseMsg)
...
+                UserDataCounter_t user_data;
+                user_data.service_name = strdup(service_name);
+                user_data.hit_count = 0;
+                user_data.delete_count = 0;
                 ParodusPrint("service_name %s\n",service_name);
-                if(true == delete_client_subscriptions(service_name))
+                delete_client_subscriptions(&user_data);
+                if(0 < user_data.delete_count)
                 {
                     ParodusInfo("%s service deleted successfully\n",service_name);
```
Comment (shilpa24balaji) [inline src/crud_tasks.c]: We can log the hit_count and delete_count

File: src/subscription.c [hunk 3]
```diff
@@ -16,6 +16,9 @@ rebar_ll_list_t *g_sub_list = NULL;
 /*----------------------------------------------------------------------------*/
 /*                             Function Prototypes                            */
 /*----------------------------------------------------------------------------*/
+static void __delete_subscritption ( rebar_ll_node_t *node, void *data );
```
Comment (shilpa24balaji) [inline src/subscription.c]: Spelling can be changed __delete_subscription

File: src/subscription.c [hunk 4]
```diff
@@ -115,23 +130,34 @@ void filter_clients_and_send(wrp_msg_t *wrp_event_msg)
...
+         Subscription *sub = rebar_ll_get_data(Subscription, sub_node, node);
+
+         if (0 == strcmp(((UserDataCounter_t *) data)->service_name, sub->service_name)) {
+             ((UserDataCounter_t *) data)->hit_count++;
+             return REBAR_IR__DELETE_AND_CONTINUE ;
+         }
     }
-    return match_found;
+
+    return REBAR_IR__CONTINUE;;
```
Comment (shilpa24balaji) [inline src/subscription.c]: Is double semicolon required here?

File: src/subscription.c (hunk 3 above)
Comment (manderiasian) [reply src/subscription.c]: Yes, I wanted to really make it clear don't export this function. We can change i on next go around.

File: src/subscription.c (hunk 4 above)
Comment (manderiasian) [reply src/subscription.c]: No. Typo.

**Pull request merged**